contracts/feature-tests/composability/interact/src/call_tree_calling_functions.rs

```diff
@@ -0,0 +1,145 @@
+use std::{cell::RefCell, rc::Rc};
+
+use forwarder_queue::QueuedCallType;
+use multiversx_sc_snippets::{
+    multiversx_sc::types::{EgldOrEsdtTokenIdentifier, EgldOrEsdtTokenPayment, MultiValueEncoded},
+    multiversx_sc_scenario::{bech32, num_bigint::BigUint, scenario_model::TxExpect, DebugApi},
+    StepBuffer,
+};
+
+use crate::{
+    call_tree::{CallNode, CallState, ForwarderQueueTarget},
+    comp_interact_controller::ComposabilityInteract,
+};
+use forwarder_queue::ProxyTrait;
+use multiversx_sc_snippets::multiversx_sc_scenario::scenario_model::IntoBlockchainCall;
+
+const FORWARD_QUEUED_CALLS_ENDPOINT: &str = "forward_queued_calls";
+const DEFAULT_GAS_LIMIT: u64 = 10_000_000;
+
+impl ComposabilityInteract {
+    pub async fn add_queued_calls_to_children(
+        &mut self,
+        forwarders: &Vec<Rc<RefCell<ForwarderQueueTarget>>>,
+        call_type: QueuedCallType,
+        endpoint_name: &str,
+        payment_token: EgldOrEsdtTokenIdentifier<DebugApi>,
+        payment_nonce: u64,
+        payment_amount: BigUint,
+    ) {
+        let mut steps = Vec::new();
+
+        for fwd_rc in forwarders {
+            let (fwd_name, fwd_children) = {
+                let fwd = fwd_rc.borrow();
+                (fwd.name.clone(), fwd.children.clone())
+            };
+            let fwd_addr = {
+                let fwd = fwd_rc.borrow();
+                fwd.address.clone().unwrap()
+            };
+            let fwd_addr_bech32 = bech32::encode(&fwd_addr);
+            let fwd_addr_expr = format!("bech32:{fwd_addr_bech32}");
+
+            for child in &fwd_children {
+                match child {
+                    CallNode::ForwarderQueue(child_fwd_rc) => {
+                        // forward_queued_calls to ForwarderQueue's children
+                        let child_fwd_addr = {
+                            let child_fwd = (*child_fwd_rc).borrow();
+                            println!("child_name: {}, parent_name: {}", child_fwd.name, &fwd_name);
+                            child_fwd.address.clone().unwrap()
+                        };
+
+                        let typed_sc_call = self
+                            .state
+                            .forwarder_queue_from_addr(&fwd_addr_expr)
+                            .add_queued_call(
+                                call_type.clone(),
+                                child_fwd_addr,
+                                DEFAULT_GAS_LIMIT,
+                                FORWARD_QUEUED_CALLS_ENDPOINT,
+                                MultiValueEncoded::<DebugApi, _>::new(),
+                            )
+                            .with_egld_or_single_esdt_transfer(EgldOrEsdtTokenPayment::new(
```

Comment:
We should go for .with_any_payment, to support multi-ESDT.

Can be done later.